tests/phpunit/integration/Core/User_Input/User_InputTest.php

```diff
@@ -197,4 +207,64 @@ public function test_set_answers() {
 			$response
 		);
 	}
+
+	public function test_set_answers__preserves_purpose_answer_user_attribution() {
+		$second_admin_id = $this->factory()->user->create( array( 'role' => 'administrator' ) );
+
+		$this->user_input->set_answers(
+			array(
+				'purpose'       => array( 'publish_blog' ),
+				'postFrequency' => array( 'daily' ),
+				'goals'         => array( 'growing_audience', 'finding_new_topics' ),
+			)
+		);
+
+		$existing_answers = $this->user_input->get_answers();
+
+		$this->assertEquals( $existing_answers['purpose']['answeredBy'], $this->user_id );
+
+		$this->user_options->switch_user( $second_admin_id );
+
+		$this->user_input->set_answers(
+			array(
+				'purpose'       => array( 'publish_blog' ),
+				'postFrequency' => array( 'weekly' ),
+				'goals'         => array( 'improving_performance' ),
+			)
+		);
+
+		$existing_answers = $this->user_input->get_answers();
+		// Since purpose answer didn't change, it should still be attributed to admin 1.
+		$this->assertEquals( $existing_answers['purpose']['answeredBy'], $this->user_id );
+	}
+
+	public function test_set_answers__assigns_correct_user_attribution_on_purpose_answer_change() {
+		$second_admin_id = $this->factory()->user->create( array( 'role' => 'administrator' ) );
+
+		$this->user_input->set_answers(
+			array(
+				'purpose'       => array( 'publish_blog' ),
+				'postFrequency' => array( 'daily' ),
+				'goals'         => array( 'growing_audience', 'finding_new_topics' ),
+			)
+		);
+
+		$existing_answers = $this->user_input->get_answers();
+
+		$this->assertEquals( $existing_answers['purpose']['answeredBy'], $this->user_id );
+
+		$this->user_options->switch_user( $second_admin_id );
+
+		$this->user_input->set_answers(
+			array(
+				'purpose'       => array( 'other' ),
+				'postFrequency' => array( 'weekly' ),
+				'goals'         => array( 'improving_performance' ),
+			)
+		);
+
+		$existing_answers = $this->user_input->get_answers();
+		// Since purpose answer changed, it should be attributed to admin 2.
+		$this->assertEquals( $existing_answers['purpose']['answeredBy'], $second_admin_id );
```

Comment:
Since we're changing all of the answers, let's check each one and make sure only the relevant ones changed attribution, just to be thorough 😄 

Basically the first question is the only site-wide one, so let's make sure the attribution for the other questions doesn't change here.